src/tr1/game/objects/traps/movable_block.c

```diff
@@ -213,6 +217,48 @@ static bool M_TestPull(ITEM *item, int32_t block_height, DIRECTION quadrant)
     return true;
 }
 
+static bool M_TestBlockDeath(ITEM *const item, ITEM *const lara)
+{
+    if (!g_GameFlow.enable_killer_pushblocks || !item->gravity
+        || lara->hit_points <= 0 || !Lara_TestBoundsCollide(item, 0)) {
+        return false;
+    }
+
+    lara->hit_points = -1;
+    lara->pos.y = lara->floor;
+    lara->speed = 0;
+    lara->fall_speed = 0;
+    lara->gravity = false;
+    lara->rot.x = 0;
+    lara->rot.z = 0;
+    lara->current_anim_state = LS_SPECIAL;
+    lara->goal_anim_state = LS_SPECIAL;
+    Item_SwitchToAnim(lara, LA_ROLLING_BALL_DEATH, 0);
+    Lara_SetShadowEnabled(false);
+
+    for (int32_t i = 0; i < 15; i++) {
+        const int32_t x = lara->pos.x + (Random_GetControl() - 0x4000) / 256;
+        const int32_t z = lara->pos.z + (Random_GetControl() - 0x4000) / 256;
+        const int32_t y = lara->pos.y - Random_GetControl() / 64;
+        const int32_t d = item->rot.y + (Random_GetControl() - 0x4000) / 8;
+        Effect_Blood(x, y, z, item->speed * 2, d, item->room_num);
+    }
+
+    if (!Object_GetObject(O_CAMERA_TARGET)->loaded) {
+        return true;
+    }
+
+    const int16_t target_num = Item_Spawn(lara, O_CAMERA_TARGET);
+    if (target_num != NO_ITEM) {
+        ITEM *const target = Item_Get(target_num);
+        target->rot.y = g_Camera.target_angle;
+        target->pos.y = lara->floor - WALL_L;
+        Lara_SetDeathCameraTarget(target_num);
+    }
+
+    return true;
```

Comment:
This function is named like a getter but has significant side effects, let's make sure the name captures it. Maybe also let's put the death logic into its own function.

Reply:
Thanks, I've moved the logic around a little and updated the naming.

Re the shadow, I initially had it so that the floor height wasn't changed but my thinking was that if there are enemies nearby, they would walk through the block. We'd also need to have something in place to update the height if Lara is resurrected (although that is a cheat so less critical).

Perhaps we could have a flag in `ITEM` for whether or not the shadow is enabled, so it would get checked in `output` for any item type?

Reply:
Thanks, those are very valid points I haven't considered. The flag sounds good, but the current approach is also fine.

Reply:
I've gone with the flag approach, which I think is a bit nearer.